client/app/screens/settings/SettingsScreen.tsx

```diff
@@ -33,18 +33,159 @@ const Sections = [
 const SettingsScreen: React.FC = () => {
   // settings values (will be changed later to reflect the actual settings)
   const [data, setData] = useState({
+    displayName: "Display Name",
+    profilePicIndex: 0, // index for icons array
+    profileColor: "#1199ff",
     notifyNewMessage: true,
     darkMode: false,
     language: "English",
     deleteMessages: false,
   });
 
+  const[profileVisible, setProfileVisible] = useState(false);
+  const[inputModal, setInputModal] = useState({
+    visible: false,
+    type: -1 // 0 for display name, 1 for profile color
+  });
+  const[textInput, setTextInput] = useState('');
+  const[errorMessage, setErrorMessage] = useState('');
+
+  const iconStyle = [styles.icon, {backgroundColor: data.profileColor}]
+
+  const icons = [
+    require("../../../assets/icons/user/face_01.png"),
+    require("../../../assets/icons/user/face_02.png"),
+    require("../../../assets/icons/user/face_03.png"),
+    require("../../../assets/icons/user/face_04.png"),
+    require("../../../assets/icons/user/face_05.png"),
+    require("../../../assets/icons/user/face_06.png"),
+    require("../../../assets/icons/user/face_07.png"),
+    require("../../../assets/icons/user/fake_pfp.jpg"),
+  ];
+
   return (
     <SafeAreaView style={styles.safeAreaStyle}>
       <ScrollView style={styles.container}>
+
+        {/* User Settings Menu */}
+        <Modal
```

Comment:
Currently, the modal isn't dismissed when the user clicks outside of it, which is a bit jarring. You can look up stuff online like [this](https://stackoverflow.com/questions/40483034/close-react-native-modal-by-clicking-on-overlay) to see how to fix that. The text input popups should have this too.

Reply:
Good point, I had meant to do this but forgot somewhere along the way. I've added it in, but for some reason I can't get the main user settings modal to fit its height to its children. I've just set its height to 50% for now, but I don't know how that looks on screens of different sizes.